decompiler/frontend/binaryninja/frontend.py

```diff
@@ -53,19 +53,21 @@ def from_raw(cls, view: BinaryView):
     def create_task(self, function: Union[str, Function], options: Options) -> DecompilerTask:
         """Create a task from the given function identifier."""
         debug_mode = options.getboolean("pipeline.debug", fallback=False)
-        if not isinstance(function, Function):
-            function = self._find_function(function)
-        return_type, params = self._extract_return_type_and_params(function)
+        if isinstance(function, Function):
+            function_obj = function
+        elif (function_obj := self._get_function_from_str(function)) is None:
+            raise ValueError(f"Could not resolve function symbol {function}")
+        return_type, params = self._extract_return_type_and_params(function_obj)
         try:
-            cfg = self._extract_cfg(function, options)
-            task = DecompilerTask(function.name, cfg, function_return_type=return_type, function_parameters=params, options=options)
+            cfg = self._extract_cfg(function_obj, options)
+            task = DecompilerTask(function_obj.name, cfg, function_return_type=return_type, function_parameters=params, options=options)
         except Exception as e:
-            task = DecompilerTask(function.name, None, function_return_type=return_type, function_parameters=params, options=options)
+            task = DecompilerTask(function_obj.name, None, function_return_type=return_type, function_parameters=params, options=options)
             task.fail(origin="CFG creation")
             logging.error(f"Failed to decompile {task.name}, error during CFG creation: {e}")
             if debug_mode:
                 raise e
-        task.function = function
+        task.function = function_obj # TODO should we get rid of monkey patching?
```

Comment:
Same as above. Honestly, I'm not even sure why we are saving this

Reply:
moved to #170